consensus/src/pipeline/pipeline_builder.rs

```diff
@@ -564,11 +564,30 @@ impl PipelineBuilder {
             user_txns.as_ref().clone(),
         ]
         .concat();
+        let proposer_index = block
+            .author()
+            .and_then(|proposer| validator.iter().position(|&v| v == proposer));
+        let extra_info = txns
+            .iter()
+            .map(|txn| {
+                txn.borrow_into_inner().try_as_signed_user_txn().map_or(
+                    ExtraInfo::new_empty(),
+                    |_| {
+                        ExtraInfo::new(
+                            PersistedExtraInfo::None,
+                            proposer_index.map(|index| EphemeralExtraInfo {
+                                proposer_index: index as u64,
```

Comment:
The proposer index is the proposer of the entire block. Why do we have to store this information for each transaction separately?

Reply:
with execution pool this assumption doesn't hold anymore.